.github/workflows/edge-preview-authenticated-proxy.yml

```diff
@@ -0,0 +1,46 @@
+name: Edge Preview Authenticated Proxy Worker
+
+on:
+  push:
+    branches:
+      - main
+jobs:
+  publish_worker:
+    if: ${{ github.repository_owner == 'cloudflare' }}
+    name: Publish Worker
+    needs: prerelease
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Use Node.js 16.13
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.13
+          cache: "npm" # cache ~/.npm in case 'npm ci' needs to run
+
+      - name: Install workerd Dependencies
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          export DEBIAN_FRONTEND=noninteractive
+          sudo apt-get update
+          sudo apt-get install -y libc++1
```

Comment:
Do we need to install workerd deps if we are not testing with `wrangler dev` in this workflow?

Reply:
We need them to build Wrangler

Reply:
Another TIL... this seems strange since we only download workerd if it is requested at runtime, right...

Reply:
Yeah—maybe I'm out of date on that requirement? I'll double check

Reply:
Ah, @mrbbot clarified—it's because `@miniflare/tre` is in the `devDependencies`